src/main/kotlin/org/rust/ide/template/postfix/stringBasedPostfixTemplates.kt

```diff
@@ -63,8 +63,14 @@ class LambdaPostfixTemplate(provider: RsPostfixTemplateProvider) : SimpleExprPos
 
 class NotPostfixTemplate(provider: RsPostfixTemplateProvider) : SimpleExprPostfixTemplate("not", "!expr", provider)
 
-class RefPostfixTemplate(provider: RsPostfixTemplateProvider) : SimpleExprPostfixTemplate("ref", "&expr", provider)
-class RefmPostfixTemplate(provider: RsPostfixTemplateProvider) : SimpleExprPostfixTemplate("refm", "&mut expr", provider)
+fun exprOrTypeSelector(): PostfixTemplateExpressionSelector {
+    return RsCombinedSelector(listOf(RsExprParentsSelector(), RsTypeParentsSelector()))
+}
+
+class RefPostfixTemplate(provider: RsPostfixTemplateProvider)
+    : SimpleExprPostfixTemplate("ref", "&expr", provider, exprOrTypeSelector())
+class RefmPostfixTemplate(provider: RsPostfixTemplateProvider)
+    : SimpleExprPostfixTemplate("refm", "&mut expr", provider, exprOrTypeSelector())
```

Comment:
Actually, I expect `&type` and `&mut type` examples in the following case instead of `&expr` and `&mut expr`
<img width="445" alt="Screen Shot 2022-01-21 at 17 42 11" src="https://user-images.githubusercontent.com/2539310/150546200-932db495-d712-4dc2-8d35-664b6f6269df.png">

So, maybe it's better just to have two different versions of `ref` and `refm` postfix templates with proper examples and selectors and get rid of `exprOrTypeSelector` at all?

Reply:
I agree, good idea.